### ISSUE TITLE
Problem with traefik log parser when Taefik is behind CloudFlare proxy

### ISSUE THREAD

**Issue by stepanov1975** [crowdsecurity/hub#554]
@AlteredCoder @he2ss @chad-jones @actions-user 
So, there is "traefik" logs parser that parses Traefik access logs. As far as I understand it takes the value of the "ClientAddr" field from the log. The problem is that my instance of Traefik is behind CloudFlare proxy, and "ClientAddr" always pointing to CloudFlare servers, which is not helpful.
Is it possible to add a configuration that will tell the parser to take a value of a different field instead of "ClientAddr"?
If a configurable field is not possible, maybe you can create traefik_cloudflare parser that will read "CF-Connecting-IP" field value instead of "ClientAddr"?

EDIT: The problem exists with JSON log format only. Remote IP have proper value if Traefik log format set is default "common"

**Comment by benscobie** [comment]
You need to add [Cloudflares IP ranges](https://www.cloudflare.com/en-gb/ips/) to forwardedHeaders.trustedIPs
https://doc.traefik.io/traefik/v2.0/routing/entrypoints/#forwarded-header

Here's mine configured via CLI:
`--entryPoints.websecure.forwardedHeaders.trustedIPs=173.245.48.0/20,103.21.244.0/22,103.22.200.0/22,103.31.4.0/22,141.101.64.0/18,108.162.192.0/18,190.93.240.0/20,188.114.96.0/20,197.234.240.0/22,198.41.128.0/17,162.158.0.0/15,104.16.0.0/13,104.24.0.0/14,172.64.0.0/13,131.0.72.0/22`

**Comment by stepanov1975** [comment]
> You need to add [Cloudflare's IP ranges](https://www.cloudflare.com/en-gb/ips/) to [forwardedHeaders.trustedIPs](https://doc.traefik.io/traefik/v2.0/routing/entrypoints/#forwarded-header) in Traefik.
> 
> Here's mine configured via CLI: `--entryPoints.websecure.forwardedHeaders.trustedIPs=173.245.48.0/20,103.21.244.0/22,103.22.200.0/22,103.31.4.0/22,141.101.64.0/18,108.162.192.0/18,190.93.240.0/20,188.114.96.0/20,197.234.240.0/22,198.41.128.0/17,162.158.0.0/15,104.16.0.0/13,104.24.0.0/14,172.64.0.0/13,131.0.72.0/22`

Did you check you see right address in the "ClientAddr" field of the log?
Because I have this and I still see inly CloudFlare addresses in the "ClientAddr" field
My configuration is below:
```
entryPoints:
  http:
    address: ":80"
    http:
      redirections:
        entryPoint:
          to: https
          scheme: https
          permanent: true
  https:
    http3: {}
    address: ":443"
    http:
      tls: 
        # Generate a wildcard domain certificate
        certResolver: cloudflare
        domains:
          - main: edited1.work
            sans:
              - '*.aedited1.work'
          - main: edited2.com
            sans:
              - '*.edited2.com'
    forwardedHeaders:
      trustedIPs:
        - "173.245.48.0/20"
        - "103.21.244.0/22"
        - "103.22.200.0/22"
        - "103.31.4.0/22"
        - "141.101.64.0/18"
        - "108.162.192.0/18"
        - "190.93.240.0/20"
        - "188.114.96.0/20"
        - "197.234.240.0/22"
        - "198.41.128.0/17"
        - "162.158.0.0/15"
        - "104.16.0.0/13"
        - "104.24.0.0/14"
        - "172.64.0.0/13"
        - "131.0.72.0/22"
        - "2400:cb00::/32"
        - "2606:4700::/32"
        - "2803:f800::/32"
        - "2405:b500::/32"
        - "2405:8100::/32"
        - "2a06:98c0::/29"
        - "2c0f:f248::/32"
```

**Comment by benscobie** [comment]
I'm not sure what you mean by the ClientAddr field. Here's an example of my access log, as you can see the first IP is a non-proxy one:
`141.98.255.151 - - [04/Oct/2022:19:38:12 +0000] "GET /build-deploy-asp-net-core-applications-jenkins-octopus-deploy/ HTTP/2.0" 200 13625 "-" "-" 27132 "benscobie@docker" "http://172.18.0.20:80" 213ms`

If I run that against cscli explain I get the following output:
![image](https://user-images.githubusercontent.com/944312/193911817-a818c7bb-e478-4456-974f-9b04afbb11d7.png)

**Comment by stepanov1975** [comment]
@benscobie 
Yes initially, the Traefik access log was set to "JSON" format, as it was in a tutorial I followed. It seems like when the format is JSON, the Traefik parser takes remote IP from the "ClientAddr" field. This field still points to the Cloudflare server even with "trustedIPs" set. After I switched to the "common" format of the Traefik access log, the remote IP changed to the proper value. It seems to me that both formats should work properly, but JSON definitely not working right. Now I am not sure if it is a problem with Traefik or the CrowdSec parser of Traefik logs.
